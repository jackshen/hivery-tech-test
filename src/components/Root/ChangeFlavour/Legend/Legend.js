import React from "react";
import styled from "styled-components";

const Key = styled.div`
  background-color: ${props => props.theme[props.color]};
  border-radius: 0.25rem;
  height: 1.375rem;
  margin: 0 0.5rem 0 1rem;
  min-width: 1.375rem;
`;

const LegendWrapper = styled.div`
  align-items: center;
  display: flex;
  grid-area: legend;
  height: fit-content;
  justify-self: end;
  padding-right: 2.8125rem;
  vertical-align: middle;
  width: fit-content;
`;

const Legend = () => (
  <LegendWrapper>
    <Key color="recommended" />
    <span>Recommended Flavours</span>
    <Key color="other" />
    <span>Other Flavours</span>
    <Key color="caution" />
    <span>Caution Flavours</span>
  </LegendWrapper>
);

export default Legend;
