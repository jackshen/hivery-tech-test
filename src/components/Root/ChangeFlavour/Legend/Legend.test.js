// including purely a snapshot test like this has its advantages and disadvantages
// the main advantage being that accidental changes to things such as the Styled Components' theme
// would not cause stylistic breaks

import React from "react";

import wrapAll from "#test/wrapAll";

import Legend from "./Legend";

describe("Legend", () => {
  test("matches snapshot", () => {
    const { container } = wrapAll(<Legend />);

    expect(container).toMatchSnapshot();
  });
});
