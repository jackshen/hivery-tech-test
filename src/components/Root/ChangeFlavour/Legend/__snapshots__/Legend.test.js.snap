// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Legend matches snapshot 1`] = `
.c1 {
  background-color: #82cc1f;
  border-radius: 0.25rem;
  height: 1.375rem;
  margin: 0 0.5rem 0 1rem;
  min-width: 1.375rem;
}

.c2 {
  background-color: #ffb107;
  border-radius: 0.25rem;
  height: 1.375rem;
  margin: 0 0.5rem 0 1rem;
  min-width: 1.375rem;
}

.c3 {
  background-color: #f65d3e;
  border-radius: 0.25rem;
  height: 1.375rem;
  margin: 0 0.5rem 0 1rem;
  min-width: 1.375rem;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  grid-area: legend;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  justify-self: end;
  padding-right: 2.8125rem;
  vertical-align: middle;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
      color="recommended"
    />
    <span>
      Recommended Flavours
    </span>
    <div
      class="c2"
      color="other"
    />
    <span>
      Other Flavours
    </span>
    <div
      class="c3"
      color="caution"
    />
    <span>
      Caution Flavours
    </span>
  </div>
</div>
`;
